hummingbot/market/kraken/kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,288 @@
+
+
+class KrakenAPIOrderBookDataSource(OrderBookTrackerDataSource):
+
+    MESSAGE_TIMEOUT = 30.0
+    PING_TIMEOUT = 10.0
+
+    _kraobds_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._kraobds_logger is None:
+            cls._kraobds_logger = logging.getLogger(__name__)
+        return cls._kraobds_logger
+
+    def __init__(self, trading_pairs: Optional[List[str]] = None):
+        super().__init__()
+        self._trading_pairs: Optional[List[str]] = trading_pairs
+        self._order_book_create_function = lambda: OrderBook()
+
+    @classmethod
+    @async_ttl_cache(ttl=60 * 30, maxsize=1)
+    async def get_active_exchange_markets(cls) -> pd.DataFrame:
+        """
+        Returned data frame should have trading_pair as index and include usd volume, baseAsset and quoteAsset
+        """
+        async with aiohttp.ClientSession() as client:
+
+            trading_pairs_response = await client.get(ASSET_PAIRS_URL)
+            trading_pairs_response: aiohttp.ClientResponse = trading_pairs_response
+
+            if trading_pairs_response.status != 200:
+                raise IOError(f"Error fetching Kraken trading pairs. "
+                              f"HTTP status is {trading_pairs_response.status}.")
+
+            trading_pairs_data: Dict[str, Any] = await trading_pairs_response.json()
+
+            trading_pairs: Dict[str, Any] = {pair: {f"{k}Asset": trading_pairs_data["result"][pair][k] for k in ["base", "quote"]}
+                                            for pair in trading_pairs_data["result"]}
+            
+            trading_pairs_str: str = ','.join(trading_pairs.keys())
+
+            market_response = await client.get(f"{TICKER_URL}?pair={trading_pairs_str}")
+            market_response: aiohttp.ClientResponse = market_response
+
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kraken markets information. "
+                              f"HTTP status is {market_response.status}.")
+
+            market_data = await market_response.json()
+
+            market_data: List[Dict[str, Any]] = [{"pair": pair, **market_data["result"][pair], **trading_pairs[pair]}
+                                                for pair in market_data["result"]
+                                                if pair in trading_pairs]
+
+            # Build the data frame.
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="pair")
+            all_markets["lastPrice"] = all_markets.c.map(lambda x: x[0])
+            quotes: Dict[str, List[str]] = {}
+            quotes["usdt"] = ["ETH", "XBT", "DAI", "USDC"]
+            quotes["usdt_r"] = ["ZUSD", "EUR", "CAD", "GBP"]
+            quotes["xeth_r"] = ["ZJPY"]
+            quotes["eth_r"] = ["CHF"]
+            quotes_all: List[str] = chain([quotes[currency] for currency in quotes])
```

Comment:
Do you think it is better to have these in a `constants.py` file, like that of the Liquid connector? I assume these values will only change only if Kraken have a new token listing.

Reply:
Good point, can do.